osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/ImportHighScoresCommand.cs

```diff
@@ -135,49 +135,67 @@ public async Task<int> OnExecuteAsync(CancellationToken cancellationToken)
 
             DateTimeOffset start = DateTimeOffset.Now;
 
-            ulong lastId;
+            ulong lastId = StartId ?? 0;
+            string where;
+            bool watchMode = !StartId.HasValue;
+            ulong maxProcessableId;
+
+            if (watchMode)
+            {
+                using var db = DatabaseAccess.GetConnection();
+
+                ulong? lastImportedLegacyScore = db.QuerySingleOrDefault<ulong?>($"SELECT MAX(old_score_id) FROM score_legacy_id_map WHERE ruleset_id = {RulesetId}") ?? 0;
+                ulong? lowestProcessQueueEntry = db.QuerySingle<ulong?>($"SELECT MIN(score_id) FROM score_process_queue WHERE is_deletion = 0 AND mode = {ruleset.RulesetInfo.OnlineID}") - 1 ?? ulong.MaxValue;
```

Comment:
I hope you've actually tested this with a "local dev setup", yeah? Because this like looks awfully weird - that it's a `QuerySingle` and not a `QuerySingleOrDefault` which looks to be more appropriate.

Reply:
I think you're correct here, will update.

Reply:
https://github.com/peppy/osu-queue-score-statistics/commit/6f7382e3d314cc3d7df848fc5468ec1c9b2c4358k